### ISSUE TITLE
BigQuery - Can't sync materialized views since 1.49.0

### ISSUE THREAD

**Issue by FilmonK** [metabase/metabase#42008]
### Describe the bug

Testing in 1.49.7.
Not able to successfully sync Materialized Views in BigQuery and run into the following error:

> java.lang.ClassCastException: class com.google.cloud.bigquery.AutoValue_MaterializedViewDefinition cannot be cast to class com.google.cloud.bigquery.StandardTableDefinition (com.google.cloud.bigquery.AutoValue_MaterializedViewDefinition and com.google.cloud.bigquery.StandardTableDefinition are in unnamed module of loader clojure.lang.DynamicClassLoader @4f5d587c)

<br>

Metadata for the materialized view within Admin Settings will not be brought in.
Attempting to query or browse the data will give a "no fields found" message. 

<img width="703" alt="Screenshot 2024-04-29 at 5 51 25 PM" src="https://github.com/metabase/metabase/assets/17398657/3d639c21-400c-4900-ae86-452d0967496a">

<br>
<br>

Tested syncing on **1.49.0** and encounter the same issue.
Syncing functions correctly **1.48.7**.

### To Reproduce

1. Create materialized view in BigQuery
2. Sync database

It doesn't matter whether this is an update to an existing materialized view, or a newly created one.

### Expected behavior

For the sync to function as it did prior to 1.49.0.

### Information about your Metabase installation

```JSON
{
  "browser-info": {
    "language": "en-US",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/124.0.0.0 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.22+7",
    "java.vendor": "Eclipse Adoptium",
    "java.vendor.url": "https://adoptium.net/",
    "java.version": "11.0.22",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.22+7",
    "os.name": "Mac OS X",
    "os.version": "13.5.1",
    "user.language": "en",
    "user.timezone": "America/Chicago"
  },
  "metabase-info": {
    "databases": [
      "postgres",
      "mongo",
      "bigquery-cloud-sdk",
      "mysql",
      "h2"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "15.3"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.7.2"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2024-04-22",
      "tag": "v1.49.7",
      "hash": "f0ff786"
    },
    "settings": {
      "report-timezone": "UTC"
    }
  }
}
```

### Severity

P1 - Not able to sync Materialized Views or query them

**Comment by Mjoubrel** [comment]
Hello Ngoc,

I noticed the issue has been addressed, thanks a lot.

Do you know when the fix will be available for Cloud users?

Best,

Mathieu Joubrel
+33 6 73 17 81 74
Book a meeting

**Comment by qnkhuat** [comment]
The fix has not been released yet, so I don't have a date for you. If you're a cloud customer, please email us so we can support you further.

**Comment by qnkhuat** [comment]
The fix will be included in 49.9. but it hasn't been released, so I don't have an exact date for you. If you're on the cloud, please email us so our success engineer can support you.
